### PR TITLE
adding k8s-cli alpha docs

### PR DESCRIPTION
**Pull request by trujillo-adam**
This PR adds a draft of the Consul K8s CLI documentation for the alpha release.
There are still some areas that need additional information.
I will open a corresponding PR to update the related tutorial(s). 

🔍  [Preview ](https://consul-47m7r06wo-hashicorp.vercel.app/docs/k8s)

### CODE REVIEW

File: website/content/docs/k8s/k8s-cli.mdx
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/k8s-cli.mdx:52]: Content missing - TODO

Comment (karl-cardenas-coding) [reply website/content/docs/k8s/k8s-cli.mdx:52]: Content missing  -TODO

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
I caught some TODOs lingering around

File: website/content/docs/k8s/installation/install.mdx
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/install.mdx:58]: The indentation is a bit off here @trujillo-adam 

image

**Review by karl-cardenas-coding — APPROVED** [review]
LGTM, just a minor indentation issue.

File: website/content/docs/k8s/k8s-cli.mdx
Comment (ndhanushkodi) [inline website/content/docs/k8s/k8s-cli.mdx:17]: We're close to finalizing the install PR into the consul-k8s repo itself, so as soon as that PR is merged, we'll want people to build it from there: (https://github.com/hashicorp/consul-k8s/tree/cli-install/cli). We should have install merged by alpha. We could totally merge this docs PR as is, and create a new docs PR to update this as soon as it's merged.

We'd just need to update this step and the step below to reference the correct paths (i.e to clone the consul-k8s repo instead, navigate to consul-k8s/cli, run go build from there.)

We may want to consider just having them do 
```
go build .
```
and adding an additional optional step to say:
(Optional) Consider moving the binary into your $PATH for ease of use:
```
mv consul-k8s /usr/local/bin/consul-k8s
```

File: website/content/docs/k8s/k8s-cli.mdx
Comment (ndhanushkodi) [inline website/content/docs/k8s/k8s-cli.mdx:50]: We decided to remove the name flag and it will be removed before alpha

File: website/content/docs/k8s/k8s-cli.mdx
Comment (ndhanushkodi) [inline website/content/docs/k8s/k8s-cli.mdx:49]: I think the -config-file flag is missing from the table https://github.com/hashicorp/consul-k8s/blob/e935f0703f63d9ad7931978ca0b3f97a2aac7b77/cli/README.md#consul-k8s-install

File: website/content/docs/k8s/k8s-cli.mdx
Comment (ndhanushkodi) [inline website/content/docs/k8s/k8s-cli.mdx:102]:
```suggestion
| `-kubeconfig` <br/> Alias: `-c`                                                             | String value that specifies the path to the `kubeconfig` file. <br/>  | none    | Optional |
```
Maybe it'll be a little easier to see aliases in the first column?

**Review by ndhanushkodi — APPROVED** [review]
This is looking so incredible Adam! There are some last minute changes to the flags in `uninstall` that I'm going to be trying to implement tonight/early tomorrow that I'll keep you posted on in the Slack channel. If it makes sense to merge this PR before those updates, and update uninstall as soon as it's more finalized that's totally fine, or if you'd like to hold this PR until uninstall is slightly more stable, that works too.

File: website/content/docs/k8s/installation/install.mdx
Comment (trujillo-adam) [reply website/content/docs/k8s/installation/install.mdx:83]: Yes, I think we can afford to wait until the flags are finalized to merge this PR.

File: website/content/docs/k8s/installation/install.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/install.mdx:22]: Could we perhaps include a TOC so one could quickly click on Helm Chart installation or Consul K8s CLI Installation?

Comment (trujillo-adam) [reply website/content/docs/k8s/installation/install.mdx:22]: The docs already have a TOC built into the UI


You think we should hard code one into the page? I believe the plan is to make the built-in TOCs more noticeable, but I don't know how far away we are from that design change.

Comment (david-yu) [reply website/content/docs/k8s/installation/install.mdx:22]: > Could we perhaps include a TOC so one could quickly click on Helm Chart installation or Consul K8s CLI Installation?

Was thinking of something like this instead. I had nod ideas there was a TOC so it was hard for a normal user to navigate quickly. 

```suggestion


Refer to the [architecture](/docs/k8s/installation/install#architecture) section to learn more about the general architecture of Consul on Kubernetes.
For a hands-on experience with Consul as a service mesh
for Kubernetes, follow the [Getting Started with Consul service
mesh](https://learn.hashicorp.com/tutorials/consul/service-mesh-deploy?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.

You can install Consul on Kubernetes in two ways: 

1. [Helm chart install](#helm-chart-installation)
2. [Consul K8s CLI install <sup>ALPHA</sup>.](#consul-k8s-cli-installation)


```

File: website/content/docs/k8s/installation/install.mdx
Comment (david-yu) [inline website/content/docs/k8s/installation/install.mdx:79]: It looks like there is an extra ` here and closing for this shell excerpt?

File: website/content/docs/k8s/operations/uninstall.mdx
Comment (david-yu) [inline website/content/docs/k8s/operations/uninstall.mdx:9]: This looks like this is the old uninstall steps for Helm, could you copy and paste from latest? It still shows deleting `kubectl delete secret consul-controller-webhook-cert` as a step which is no longer needed.

Comment (trujillo-adam) [reply website/content/docs/k8s/operations/uninstall.mdx:9]: Yeah, I'm having some git trouble, trying to get things into the correct state

Comment (trujillo-adam) [reply website/content/docs/k8s/operations/uninstall.mdx:9]: Oh, I thought you were referring to some content regressions (I had some conflicts earlier), but you literally meant that this content needs to be updated. Where do I get this information? I looked at the Kubernetes docs and this appears to still be the correct usage: https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#delete

Comment (david-yu) [reply website/content/docs/k8s/operations/uninstall.mdx:9]: Yes we should use `helm uninstall` instead of `helm delete`. 

I was saying that https://www.consul.io/docs/k8s/operations/uninstall should have the latest correct uninstall process that I merged yesterday, but yea I meant the content had old content from a previous version of the uninstall guide. We also merge 3 and 4 into one bullet as well since they are part of the same step.

File: website/content/docs/k8s/k8s-cli.mdx
Comment (david-yu) [inline website/content/docs/k8s/k8s-cli.mdx:25]: Would it be possible to indent this section here?

File: website/content/docs/k8s/operations/uninstall.mdx
Comment (david-yu) [inline website/content/docs/k8s/operations/uninstall.mdx:84]:
```suggestion
Issue the `consul-k8s uninstall` command to remove Consul on Kubernetes. You can specify the installation name, namespace, and data retention behavior using the applicable options. By default, the uninstallation preserves the secrets and PVCs that are provisioned by Consul on Kubernetes. 
```

File: website/content/docs/k8s/operations/uninstall.mdx
Comment (david-yu) [inline website/content/docs/k8s/operations/uninstall.mdx:90]: You could move this above to make it more readable. 
```suggestion
```

**Review by david-yu — APPROVED** [review]
LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/455886.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit b22684540eb913243bc63228182f4d3bae18b0e4 onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/455886)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit b22684540eb913243bc63228182f4d3bae18b0e4 onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/455886)

**Pull request merged**
